libvast/vast/detail/env.hpp

```diff
@@ -0,0 +1,26 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <optional>
+#include <string>
+
+namespace vast::detail {
+
+/// A thread-safe wrapper around `::getenv`.
+/// @param var The environment variable.
+/// @returns The copied environment variables contents, or `std::nullopt`.
+[[nodiscard]] std::optional<std::string> env(const char* var);
```

Comment:
I'd propose we stay close to the wrapped functions and call these `getenv/unsetenv`? Or maybe even something like `getenv_mt`/`locked_getenv`/`blocking_getenv`/`threadsafe_getenv` to make it clear at the callsite why we don't just use the standard library functions.

Reply:
I like the `locked_` prefix, will change.